Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,22 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python?(python)
+        return true if build.with?(python)
+        (requirements.to_a | deps).any? { |r| r.name == python && r.required? }
+      end
+
       # Helper method for the common case of installing a Python application.
       # Creates a virtualenv in `libexec`, installs all `resource`s defined
       # on the formula, and then installs the formula.
-      def virtualenv_install_with_resources
-        venv = virtualenv_create(libexec)
+      def virtualenv_install_with_resources(options = {})
```

Comment:
I agree almost universally. I'm not a big fan of `:python_exe` as it smells too much like Windows, but `:python_binary` or something similar would be fine. Maybe to avoid the slightly bogus look of `:python => "python"` at the cost of hard-coding binary names we could make that `:version => 2` or `python_version => 2` or `python_major => 2` (naming is hard!) and leave mapping that to an appropriate Python binary name to the method.

One major downside would be the confusingly different interface of `virtualenv_create` and this method, so it might be prudent to adjust the former if we decide to tweak the latter.